solutions/coci-21-sjeckanje.mdx

```diff
@@ -0,0 +1,169 @@
+---
+id: coci-21-sjeckanje
+source: COCI 2021
+title: Sjeckanje
+author: Timothy Gao
+---
+
+For any segment $[l, r]$ with more than three elements, if there exists a tuple such that $a_i \leq a_{i+2} \leq a_{i+1}$ or $a_i \geq a_{i+2} \geq a_{i+1}$ where $i$ to $i+2$ are contained in the segment, then the segment can be split into two segments $l, l+1, \dots, i+1$ and $i+2, i+3, \dots, r$. This is provably at least as good as the score ($\text {max} - \text {min}$) of the initial segment. If the segment isn't strictly increasing or strictly decreasing, there must be three consecutive elements within the segment that aren't in strictly increasing or decreasing order, and the first two elements of the segment determine whether the rest of the segment will be strictly increasing or decreasing. Therefore, an optimal solution can be constructed with only monotonically increasing or decreasing segments. This paves the way for several key properties.
```

Comment:
```suggestion
For any segment $[l, r]$ with at least three elements, if there exists a tuple such that $a_i \leq a_{i+2} \leq a_{i+1}$ or $a_i \geq a_{i+2} \geq a_{i+1}$ where $i$ to $i+2$ are contained in the segment, then the segment can be split into two segments $l, l+1, \dots, i+1$ and $i+2, i+3, \dots, r$. This is provably at least as good as the value of the initial segment ($\text {max} - \text {min}$). More generally, if a segment isn't strictly increasing or strictly decreasing, we can split it into two segments without lowering the total value. Therefore, an optimal solution can be constructed with only strictly increasing or decreasing segments. This paves the way for several key properties.
```

Reply:
monotonic != strict btw https://en.wikipedia.org/wiki/Monotonic_function#In_calculus_and_analysis

Reply:
My argument initially was that the solution could be constructed with monotonously non-decreasing or non-increasing segments, but just having it be strictly increasing or decreasing makes the explanation more clear. Thanks for the suggestions.